print_nanny_webapp/alerts/views.py

```diff
@@ -65,6 +72,39 @@ def defect_form_valid(self, form):
         success_url = self.get_success_url()
         return HttpResponseRedirect(success_url)
 
+    def create_discord_form(self, **kwargs):
+        instance, created = DiscordMethodSettings.objects.get_or_create(
+            user=self.request.user,
+        )
+        if instance is not None:
+            return DiscordMethodSettingsForm(instance=instance, **kwargs)
+        else:
+            return DiscordMethodSettingsForm(**kwargs)
+
+    def discord_form_valid(self, form):
+        obj = form.save(commit=False)
+
+        channel_layer = get_channel_layer()
+        res = async_to_sync(channel_layer.send)(
+            "discord", {
+                "type": "validate.id",
+                "target_id": obj.target_id,
+                "target_id_type": obj.target_id_type,
+        })
+
+        if res is None:
+            raise ValidationError("Could not validate Discord ID - Maybe Discord is down?")
+
+        if not res["is_valid"]:
+            raise ValidationError(res["error"])
```

Comment:
@leigh-johnson this is where `validate.id` should be called. I'm validating that the given IDs are supported and actually reachable

Reply:
Ahhh I see, thanks for pointing me to this. 

Whoops, I didn't think through how the Discord response would be received in the view. 

From the Channel layer, the mechanism to "send" data from the channel consumer is usually via a connected websocket client. Buuuuut the design pattern I linked you to works for one-way communication (sending work to a background task). 

Are you ok with punting this for now?

Sorry for sending you down a rabbit hole!